### PR TITLE
[FLINK-4506] CsvOutputFormat defaults allowNullValues to false, even …

### PR DESCRIPTION
**Pull request by kirill-morozov-epam**

### CODE REVIEW

**Comment by zentol** [comment]
I think should rather adjust the documentation to fit the current behavior. This discrepancy wasn't just introduced recently, as far as i can tell it has been there forever, or rather since the allowNullValues flag was introduced.

**Comment by fhueske** [comment]
I agree with @zentol. `this.allowNullValues = false;` is the behavior since the initial code import. I would not change this and rather fix the JavaDocs.

File: flink-java/src/test/java/org/apache/flink/api/java/io/CsvOutputFormatTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	}
+
+	@Test(expected = RuntimeException.class)
+	public void testNullDisallowOnDefault() throws Exception {
+		CsvOutputFormat<Tuple3<String, String, Integer>> csvOutputFormat = new CsvOutputFormat<Tuple3<String, String, Integer>>(PATH);
+		csvOutputFormat.setWriteMode(FileSystem.WriteMode.OVERWRITE);
+		csvOutputFormat.setOutputDirectoryMode(FileOutputFormat.OutputDirectoryMode.PARONLY);
+		csvOutputFormat.open(0, 1);
+		csvOutputFormat.writeRecord(new Tuple3<String, String, Integer>("One", null, 8));
+		csvOutputFormat.close();
```
Comment (zentol) [inline flink-java/src/test/java/org/apache/flink/api/java/io/CsvOutputFormatTest.java]: The temporary file is created within `open()` and will not be cleaned up.

Comment (kirill-morozov-epam) [inline flink-java/src/test/java/org/apache/flink/api/java/io/CsvOutputFormatTest.java]: Fixed in https://github.com/apache/flink/pull/2477/commits/6d06d0e03e12ced8fe11e164488f1102682704d1

File: flink-java/src/test/java/org/apache/flink/api/java/io/CsvOutputFormatTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class CsvOutputFormatTest {
+
+	private static final Path PATH = new Path("csv_output_test_file.csv");
```
Comment (fhueske) [inline flink-java/src/test/java/org/apache/flink/api/java/io/CsvOutputFormatTest.java]: Please use `File.createTempFile()` to create a file in the default temp space.

**Comment by fhueske** [comment]
Thanks for the update @kirill-morozov-epam!
Merging

**Pull request closed without merging**